### PR TITLE
jonna's movies

### PR DESCRIPTION
**Pull request by jonnaru**
Link to my deployed project:
https://jonnas-movies.netlify.app/

My reflections on how this week's project turned out:
It is so much fun when you can use information from the API!!! 👩‍💻

Things I'd like to have clarified or explained in more detail:
I think this week has been very clear ☀️

### CODE REVIEW

**Review by liza-minican — APPROVED** [review]
Jonna, I am really impressed by the amount of effort you've put into the project!! It is Black level and looks super professional! Your code is clear and functional as well. Don't know what to add here, only will ask you for a favor to explain me several things when you have time:)

File: code/src/components/Nav.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+import useDetectOutsideClick from "./UseDetectOutsideClick";
+import "./nav.css";
+
+// I wanted to do a dropdown, found this "dropdownRef"-code below and tryed it out!
+
+const Nav = () => {
+  const dropdownRef = useRef(null);
+  const [isActive, setIsActive] = useDetectOutsideClick(dropdownRef, false);
+  const onClick = () => setIsActive(!isActive);
```
Comment (EmOlsz) [inline code/src/components/Nav.js:12]: A bit more information in function name would be very nice. At least let other developers when will this function be executed : `const onButtonClick = () => { ... }`

File: code/src/components/UseDetectOutsideClick.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+// FOUND THIS CODE. WANT TO LEARN MORE ABOUT THIS:
+
+/**
+ * Hook for handling closing when clicking outside of an element
+ * @param {React.node} el
+ * @param {boolean} initialState
+ */
+
+const UseDetectOutsideClick = (el, initialState) => {
```
Comment (EmOlsz) [inline code/src/components/UseDetectOutsideClick.js:11]: Custom hook, many congratulations! Tremendous job :)

File: code/src/pages/MovieDetail.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+              <div>
+                <h1>
+                  {movie.title}
+                  <span className="movie-detial-rating">
+                    {movie.vote_average}/10
+                  </span>
+                </h1>
+                <p className="credits-name">
+                  {credits.cast.slice(0, 5).map((actor, index) => (
+                    <>
```
Comment (EmOlsz) [inline code/src/pages/MovieDetail.js:91]: Missing `key` attribute. We should avoid React fragment elements inside loops for one reason - we are not able to add any attributes to them. Approach below will be more appropriate :

```
<div key={actor.id}>
  <Link to={`/person/${actor.id}`}>{actor.name}</Link>
  <span>{punctuation(index, 5)}</span>
</div>
```

File: code/src/pages/Person.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        return res.json();
+      })
+      .then((json) => {
+        setPerson(json);
+        setTimeout(() => {
+          setLoading(false);
+        }, 500);
+      })
+      .catch((error) => {
+        console.log("error", error);
```
Comment (EmOlsz) [inline code/src/pages/Person.js:32]: Once again, `console.error()` will be more appropriate here

**Comment by EmOlsz** [comment]
Great job Jonna! ⭐️

**Pull request closed without merging**